Testscripts/Windows/PERF-STORAGE-FIO-HB.ps1

```diff
@@ -0,0 +1,316 @@
+				} else {
+					Write-LogInfo "SetupHbKernel.sh finished on $($AllVMData.RoleName)"
+					$vmCount--
+				}
+				break
+			} elseif ($state -eq "TestSkipped") {
+				Write-LogInfo "SetupHbKernel.sh finished with SKIPPED state!"
+				$resultArr = $resultSkipped
+				$currentTestResult.TestResult = Get-FinalResultHeader -resultarr $resultArr
+				return $currentTestResult.TestResult
+			} elseif ($state -eq "TestFailed") {
+				Write-LogErr "SetupHbKernel.sh didn't finish successfully!"
+				$resultArr = $resultFail
+				$currentTestResult.TestResult = Get-FinalResultHeader -resultarr $resultArr
+				return $currentTestResult.TestResult
+			} elseif ($state -eq "TestAborted") {
+				Write-LogInfo "SetupHbKernel.sh finished with Aborted state!"
+				$resultArr = $resultAborted
+				$currentTestResult.TestResult = Get-FinalResultHeader -resultarr $resultArr
+				return $currentTestResult.TestResult
+			} else {
+				Write-LogInfo "SetupHbKernel.sh is still running in the VM!"
+			}
+		}
+		if ($vmCount -le 0){
+			Write-LogInfo "SetupHbKernel.sh is done"
+		} else {
+			throw "SetupHbKernel.sh didn't finish in the VM!"
+		}
+
+		# Reboot VM to apply swap setup changes
+		Write-LogInfo "Rebooting All VMs!"
+		$TestProvider.RestartAllDeployments($AllVMData)
+
+		# Check the VM status before hibernation
+		$vmStatus = Get-AzVM -Name $vmName -ResourceGroupName $rgName -Status
+		if ($vmStatus.Statuses[1].DisplayStatus = "VM running") {
+			Write-LogInfo "$($vmStatus.Statuses[1].DisplayStatus): Verified successfully VM status is running before hibernation"
+		} else {
+			Write-LogErr "$($vmStatus.Statuses[1].DisplayStatus): Could not find the VM status before hibernation"
+			throw "Can not identify VM status before hibernate"
+		}
+
+		# Send fio 32-job command for 5 min
+		Write-LogInfo "Running fio-1 command"
+		$timeout = New-Timespan -Minutes $maxFIORunWaitMin
+		$sw = [diagnostics.stopwatch]::StartNew()
+		$state = Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "bash ./fio1Command.sh" -RunInBackground -runAsSudo
+		Wait-Time -seconds 5
+		while ($sw.elapsed -lt $timeout){
+			Wait-Time -seconds 15
+			$state = Run-LinuxCmd -ip $VMData.PublicIP -port $VMData.SSHPort -username $user -password $password "cat /home/$username/state.txt"
+			if ($state -eq "TestCompleted") {
+				Write-LogInfo "Completed fio command execution in the VM $($AllVMData.RoleName) successfully"
+				break
+			} else {
+				Write-LogInfo "$($state): fio command is still running!"
+			}
+		}
+		if ($state -ne "TestCompleted") {
+			throw "fio-1 command is still running after $maxFIORunWaitMin minutes"
+		}
+
+		# Hibernate the VM
+		Write-LogInfo "Hibernating ..."
+		Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "./test.sh" -runAsSudo -RunInBackground -ignoreLinuxExitCode:$true | Out-Null
```

Comment:
what is test.sh?

Reply:
It is generated in SetupHbKernel.sh. hibernation command. This script generation is independent of kernel compilation.

Reply:
Fixed differently in 1fbc3c0e4f301ee22518072fd3e2d1d4bd93a1ed. Move the test.sh generation to PS1 script.